system/security/shared/caching.cpp

```diff
@@ -79,14 +79,14 @@ int CResPermissionsCache::lookup( IArrayOf<ISecResource>& resources, bool* pFoun
         if (it != m_resAccessMap.end())//exists in cache
         {
             ResPermCacheEntry& resParamCacheEntry = (*it).second;
-            const time_t timestamp = resParamCacheEntry.first;
+            const time_t timeExpiry = resParamCacheEntry.first + m_pParentCache->getCacheTimeout();
 
-            if (timestamp < tstamp)//entry was not stale during last cleanup but is stale now
+            if (timeExpiry < tstamp)//entry was not stale during last cleanup but is stale now
                 *pFound++ = false;
             else if(!m_pParentCache->isCacheEnabled() && m_pParentCache->isTransactionalEnabled())//m_pParentCache->getOriginalTimeout() == 0)
             {
                 time_t tctime = getThreadCreateTime();
-                if(tctime <= 0 || timestamp < tctime)
+                if(tctime <= 0 || timeExpiry < tctime)
```

Comment:
Not sure I follow the logic here, but it looks like this branch is used when the cache is not enabled - so I'm not sure why the cache timeout should be a factor in deciding what to do

Reply:
The test for the parent cache (CPermissionsCache) being disabled seems odd, but I don't want to change it in a service release. Opened https://track.hpccsystems.com/browse/HPCC-20121 to investigate
@richardkchapman Please merge